common/persistence/persistence_interface.go

```diff
@@ -161,8 +161,8 @@ type (
 		ForkHistoryBranch(ctx context.Context, request *InternalForkHistoryBranchRequest) error
 		// DeleteHistoryBranch removes a branch
 		DeleteHistoryBranch(ctx context.Context, request *InternalDeleteHistoryBranchRequest) error
-		// GetHistoryTree returns all branch information of a tree
-		GetHistoryTree(ctx context.Context, request *InternalGetHistoryTreeRequest) (*InternalGetHistoryTreeResponse, error)
+		// GetHistoryTreeContainingBranch returns all branch information of a tree containing the requested branch
+		GetHistoryTreeContainingBranch(ctx context.Context, request *InternalGetHistoryTreeContainingBranchRequest) (*InternalGetHistoryTreeContainingBranchResponse, error)
```

Comment:
`GetHistoryTreesWithBranch` looks better to me.
And comment should say: "GetHistoryTreesWithBranch returns all trees with specified branch".

Reply:
I don't have strong opinion here but I still think containing can cause less confusion here. 
`With` is unclear it's referring to the request or the response I think?

I will stick with `Containing` for now. Let me know if you have strong opinion on this, and I can rename.

re. the comment, only one tree will contain the requested branch and will be returned.